src/asm/opt.c

```diff
@@ -71,16 +78,31 @@ void opt_Parse(char *s)
 			error("Must specify exactly 4 characters for option 'g'\n");
 		break;
 
+	case 'f':
+		if (strlen(&s[1]) <= 2) {
+			int result;
+			unsigned int precision;
+
+			result = sscanf(&s[1], "%u", &precision);
+			if ((result != EOF && result != 1) || precision < 1 || precision > 31)
+				error("Invalid argument for option 'f'\n");
```

Comment:
That's not very descriptive of what the error is.

Reply:
I'll add a separate out-of-range message for being outside [1, 31].